presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPlanRemoteProjections.java

```diff
@@ -0,0 +1,264 @@
+            "",
+            Optional.empty());
+
+    public static final SqlInvokedFunction FUNCTION_REMOTE_FOO_1 = new SqlInvokedFunction(
+            REMOTE_FOO,
+            ImmutableList.of(new Parameter("x", parseTypeSignature(StandardTypes.INTEGER))),
+            parseTypeSignature(StandardTypes.INTEGER),
+            "remote_foo(x)",
+            RoutineCharacteristics.builder().setLanguage(JAVA).setDeterminism(DETERMINISTIC).setNullCallClause(RETURNS_NULL_ON_NULL_INPUT).build(),
+            "",
+            Optional.empty());
+
+    public static final SqlInvokedFunction FUNCTION_REMOTE_FOO_2 = new SqlInvokedFunction(
+            REMOTE_FOO,
+            ImmutableList.of(new Parameter("x", parseTypeSignature(StandardTypes.INTEGER)), new Parameter("y", parseTypeSignature(StandardTypes.INTEGER))),
+            parseTypeSignature(StandardTypes.INTEGER),
+            "remote_foo(x, y)",
+            RoutineCharacteristics.builder().setLanguage(JAVA).setDeterminism(DETERMINISTIC).setNullCallClause(RETURNS_NULL_ON_NULL_INPUT).build(),
+            "",
+            Optional.empty());
+
+    public static final SqlInvokedFunction FUNCTION_REMOTE_FOO_3 = new SqlInvokedFunction(
+            REMOTE_FOO,
+            ImmutableList.of(new Parameter("x", parseTypeSignature(StandardTypes.INTEGER)), new Parameter("y", parseTypeSignature(StandardTypes.INTEGER)), new Parameter("z", parseTypeSignature(DOUBLE))),
+            parseTypeSignature(StandardTypes.INTEGER),
+            "remote_foo(x, y, z)",
+            RoutineCharacteristics.builder().setLanguage(JAVA).setNullCallClause(RETURNS_NULL_ON_NULL_INPUT).build(),
+            "",
+            Optional.empty());
+
+    private static final Metadata METADATA = MetadataManager.createTestMetadataManager();
+
+    public TestPlanRemoteProjections()
+    {
+        FunctionManager functionManager = METADATA.getFunctionManager();
+        functionManager.addTestFunctionNamespace("unittest", new InMemoryFunctionNamespaceManager("unittest", new SqlInvokedFunctionNamespaceManagerConfig().setSupportedFunctionLanguages("sql,java")));
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_0, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_1, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_2, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_3, true);
+    }
+
+    @BeforeClass
+    public void setup()
+    {
+        FunctionManager functionManager = getFunctionManager();
+        functionManager.addTestFunctionNamespace("unittest", new InMemoryFunctionNamespaceManager("unittest", new SqlInvokedFunctionNamespaceManagerConfig().setSupportedFunctionLanguages("sql,java")));
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_0, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_1, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_2, true);
+        functionManager.createFunction(FUNCTION_REMOTE_FOO_3, true);
```

Comment:
Those exact setups have already been done in the constructor.

Reply:
The two types of tests are setup differently. They are not using the same function manager. I don't remember the specifics. If you really care I can see what I can do about it.